sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -146,6 +161,23 @@ def load(*args, **kwargs) -> "AzureAppConfigurationProvider":
     if (endpoint or credential) and connection_string:
         raise ValueError("Please pass either endpoint and credential, or a connection string.")
 
+    # Removing use of AzureAppConfigurationKeyVaultOptions
+    if (
+        "keyvault_credential" not in kwargs
+        and "secret_resolver" not in kwargs
+        and "keyvault_client_configs" not in kwargs
+        and key_vault_options
```

Comment:
This logic seems odd - it basically will silently ignore `key_vault_options` if any of the new kwargs are provided?
I think it should just be:
```python
if key_vault_options:
```
As we're already doing the correct overwrites on lines 171-176.

The other option is simply to raise a ValueError if both `key_vault_options` and the new kwargs are used together. That shouldn't break any existing customers. What do you think @xiangyan99?

Reply:
@annatisch @xiangyan99 and I already had this disscussion in the other direction https://github.com/Azure/azure-sdk-for-python/pull/31518#discussion_r1337862421